### DISCUSSION TITLE
Onchain deal renewal for deal sectors

### DISCUSSION THREAD

**Discussion by flyworker** [filecoin-project/FIPs#343] [Enhancements - Technical]
Currently cc sectors can be extended when it close to expire and can support as long as 5 years.
However the deal sector cannot be extend, for enterprise usage, 540 days is not enough.
Is it possible to do deal renewal for data deals?

Would like to know the technical effort and the time needed for implementation.

**Comment by kaitlin-beegle** [comment]
Hey @flyworker! Would you mind providing some more information about the use-case for this change? In other words, what benefit is there to extending deal sectors beyond the current 540 day limit?

**Comment by flyworker** [comment]
Let's imaging we have a business client want to have data stored for 3 years.
With the current 540 days limit, we cannot sign the contract with them. Because:

When 540 days limit approaches, we need to 

- retrieval the data,but we do not know the price for retrieval at the moment
- resealing data. We need spend a lot of resource to re-sealing the data

Even we know the price, it still a double investment if can can make a extended contract.

**Comment by anorth** [comment]
Thanks for asking, I understand the desire and am working towards it.

Deals of longer length will be possible after the changes described in #313 (decouple FIL+ from deals), and, even moreso after #298 (support third-party storage markets). But enabling the FVM is currently higher priority, so detailed design and implementation work on those proposals is currently waiting for FVM to be closer to launch.

It would be possible to do custom work in the built-in actors prior to this, but I think we should do this within the FVM version of the built-in actors rather than before that (which would slow down the FVM timelines).

**Comment by flyworker** [comment]
@anorth  thanks for replying. I was telling there is no FIP or roadmap for this issue https://filecoinproject.slack.com/archives/C016APFREQK/p1638416597105700, so I am worried.

Now I feel much more comfortable, will it possible to be implemented in next 18 months?

**Reply by anorth** [reply]
I expect so, yes.

**Reply by kaitlin-beegle** [reply]
@flyworker if you would like to author the FIP, we can also support you in doing so!
